Scripts/GameLoader.gd

```diff
@@ -0,0 +1,175 @@
+extends Node
+
+signal save_data_was_loaded
+
+const SAVE_PATH = "res://save.json"
+
+const OFFLINE_LIMIT : bool = false
+const OFFLINE_LIMIT_TIME : int = 7200 
+
+const OFFINE_BONUS_GOLD_RATIO : float = 0.3 
+const OFFINE_BONUS_XP_RATIO : float = 0.2
+
+var level_manager
+var elf
+var publican
```

Comment:
We can change this three variables into:

```GDScript
onready var publican = get_parent().get_node("World").find_node("Publican")
```

Reply:
I tried to, but GameLoader is a singleton and `get_node("World")` never  succeeds, because World doesn't exist before loading singletons. Previously they were loaded in World's `_ready()` function. So I think I have to leave them in `setup()` func.

Reply:
OK, I see. Leave it as is then.